packages/ckeditor5-core/src/editor/utils/dataapimixin.ts

```diff
@@ -19,7 +19,7 @@ import type { Constructor, Mixed } from '@ckeditor/ckeditor5-utils';
  */
 export default function DataApiMixin<Base extends Constructor<Editor>>( base: Base ): Mixed<Base, DataApi> {
 	abstract class Mixin extends base implements DataApi {
-		public setData( data: string ): void {
+		public setData( data: string | Record<string, string> ): void {
```

Comment:
This goes in line with `EditorConfig#initialData`. Alternatively, `MultiRootEditor` may not extend `DataApiMixin` and implement these methods itself. I don't remember if that caused any problems (whether editor instance is used as `DataApi`).

But in this case, this is more safe, as we broaden the input parameter, and, as I said, `data.set()` already handles this type too.